### PR TITLE
Add deprecation warnings and docstring pointers to flax.training.lr_schedule

### LINKED ISSUE #2699: Add a deprecation warning to `flax.training.lr_schedule`
The package should be deprecated in favor of Optax optimizer schedules: https://optax.readthedocs.io/en/latest/api.html#optimizer-schedules

It would help to add a warning in docstring or in code block for the existed flax learning rate schedules, to remind people to switch.

### PR DESCRIPTION
**Pull request by IvyZX**
Fixes #2699

## Checklist
- [x] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [ ] This change is discussed in a Github issue/
      [discussion](https://github.com/google/flax/discussions) (please add a
      link).
- [x] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [ ] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

### CODE REVIEW

**Review by marcvanzee — APPROVED** [review]
Thank you!
